app/controllers/shared/api.php

```diff
@@ -487,22 +484,17 @@
                     $resourceType = $parseLabel($pattern, $responsePayload, $requestParams, $user);
                 }
 
-                $key = md5($request->getURI() . implode('*', $request->getParams())) . '*' . APP_CACHE_BUSTER;
-                $data = json_encode([
-                'resourceType' => $resourceType,
-                'resource' => $resource,
-                'contentType' => $response->getContentType(),
-                'payload' => base64_encode($data['payload']),
-                ]) ;
-
-                $signature = md5($data);
-                $cacheLog  = $dbForProject->getDocument('cache', $key);
+                $key = md5($request->getURI() . '*' . implode('*', $request->getParams())) . '*' . APP_CACHE_BUSTER;
+                $signature = md5($data['payload']);
+                $cacheLog  =  Authorization::skip(fn () => $dbForProject->getDocument('cache', $key));
                 $accessedAt = $cacheLog->getAttribute('accessedAt', '');
                 $now = DateTime::now();
                 if ($cacheLog->isEmpty()) {
                     Authorization::skip(fn () => $dbForProject->createDocument('cache', new Document([
                     '$id' => $key,
                     'resource' => $resource,
+                    'resourceType' => $resourceType,
```

Comment:
`$resourceType` looks like it's the `$id` of the bucket, but should we use the internal id instead? same with `$resource`.

Reply:
We also use bucketId and fileId in getDocument() to check cache file permissions.

Reply:
I see, we get the IDs from the cacheLog:

https://github.com/appwrite/appwrite/blob/2f7b0c9938e55cae647ac72651d2f91574843342/app/controllers/shared/api.php#L211-L216

We may end up deleting the wrong cache in cases where the delete job runs after a file/bucket was recreated with the same ID, but it's just a cache, so it's okay.